samples/generic-synthetic-broken-links/brokenLinkChecker.js

```diff
@@ -0,0 +1,136 @@
+const puppeteer = require('puppeteer');
+
+/**
+ * Creates a dictionary of links that have a non-2xx status code (mapped to their respective status codes).
+ * @param {Array} An array of link objects with targetUrl and statusCode properties.
+ * @throws {Error} If any links have a non-2xx status code.
+ */
+function createNon2xxLinksDictionary(linksResult) {
+  const non2xxLinks = {};
+
+  for (const link of linksResult) {
+    const statusCode = link.statusCode;
+    const targetUrl = link.targetUrl;
+
+    if (statusCode < 200 || statusCode >= 300) {
+      non2xxLinks[targetUrl] = statusCode;
+    }
+  }
+
+  if (Object.keys(non2xxLinks).length === 0) {
+    return;
+  } else {
+    const stringified_dict = Object.entries(non2xxLinks).map(([key, value]) => `${key}:${value}`).join(", ");
```

Comment:
I would add new lines to this error message with each link. Anything else we can add, status code per link?

Reply:
this stringify happens after all links have been added so the final error message currently looks like:
"Non 2xx Links: {'abcdas': 404,  'avcad': 304}"
which I believe is what we are aiming for?

Reply:
I was thinking more like...
"Non 2xx Links:
  abcdas: 404
  avcad: 304
  asdfd: invalid link // this is for one of the non-response code examples"

etc. 

think about it this way, the error message will be shown the user and I think it should be closer to a human readable format.

Reply:
done. remove curly brackets and added line breaks between every link